hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/health/HealthCheckHelpers.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.hmpps.kotlin.health
+
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.boot.actuate.health.ReactiveHealthIndicator
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+
+abstract class HealthCheck(private val webClient: WebClient) : HealthIndicator {
+  override fun health(): Health = webClient.health()
+    .block() ?: Health.down().withDetail("HttpStatus", "No response returned from ping").build()
```

Comment:
I know this is the implementation we have in all our services, but I think it evolved without us ever changing the name.  It is called `health()` but does a `ping` instead.  This is confusing - as it looks like it should be calling `/health` to check the health of the dependencies, but it doesn't (to prevent cascading health checks).
In terms of the library I think it would be better to be called `PingCheck` and method called `WebClient.ping()` so that people can tell what it is actually doing.